src/src/PluginStructs/P163_data_struct.cpp

```diff
@@ -0,0 +1,110 @@
+#include "../PluginStructs/P163_data_struct.h"
+
+#ifdef USES_P163
+
+/**************************************************************************
+* Constructor
+**************************************************************************/
+P163_data_struct::P163_data_struct(struct EventStruct *event) {
+  _lowPowerMode = P163_GET_LOW_POWER;
+  _ledState     = P163_GET_LED_STATE;
+  _threshold    = P163_CFG_THRESHOLD;
+  _changeOnly   = _threshold < 0;
+}
+
+P163_data_struct::~P163_data_struct() {
+  delete sensor;
+}
+
+/*****************************************************
+* init
+*****************************************************/
+bool P163_data_struct::init(struct EventStruct *event) {
+  sensor = new (std::nothrow) CG_RadSens(RS_DEFAULT_I2C_ADDRESS);
```

Comment:
Should check if `sensor !- nullptr`
If not, then call `delete sensor;`

Otherwise you will have a memory leak if init is called more than once.

Reply:
Fixed it